### ISSUE TITLE
Missing JavaFx

### ISSUE THREAD

**Issue by Chrisspy36** [Querz/mcaselector#379]
**Describe the bug**
When I try to download the latest version 2.1.jar universal on Mac I keep getting a message that says I am missing JavaFX. I recently just downloaded java 18 and read a bunch of stuff on what to try but none of the fixes seem to be using the 2.1 version and only refer to the 2.0.2 version. I am very new to all of this so if someone could explain this to me like you would a 4 year old that would be extremely helpful! I have never tried to mess with code or anything like that up until today. So please be understanding. 

**Environment (please complete the following information):**
- OS: macOS Monterey Version 12.4
- Java version: 18
- Version of MCA Selector 2.1

**Comment by doinkythederp** [comment]
Here's what worked for me:

## Downloading JavaFX and the app

First you'll need to download JavaFX from this website: https://gluonhq.com/products/javafx/
![Screen Shot 2022-09-05 at 11 00 12 AM](https://user-images.githubusercontent.com/75394401/188498159-21ca2c3e-4d4d-4fdb-bdcc-c407ffe0a7a5.png)
Set the dropdown menus to what's shown above to get the correct download. Use the "aarch64" architecture if you're using an M1 Mac, and "x64" if you're using an Intel Mac.

![Screen Shot 2022-09-05 at 11 03 06 AM](https://user-images.githubusercontent.com/75394401/188498435-2e0ae22f-616c-4cde-bec9-9e293ddbfbbf.png)
After downloading it, you should end up with a zip file called something like `openjfx-18.0.2_osx-aarch64_bin-sdk.zip`. Unzip it if you haven't already, giving you the folder `javafx-sdk-18.0.2`. Open that folder, and keep it open through the next steps.

Then, download MCA Selector 2.1 (Universal), giving you a file called "mcaselector-2.1.jar" in your Downloads folder. 

## Starting the app

Now, use Command+Space to open Spotlight, type "Terminal" and press enter to open the application. Type the following in order, separated by spaces (it's very important that you get every letter and space right; double check before continuing):

1. `java --module-path`
2. Drag and drop the folder called `lib` inside the `javafx-sdk-18.0.2` folder into the Terminal.
3. `--add-modules ALL-MODULE-PATH -jar`
4. Drag and drop `mcaselector-2.1.jar` in your Downloads folder into the Terminal.

![Screen Shot 2022-09-05 at 11 35 27 AM](https://user-images.githubusercontent.com/75394401/188501030-5cf00092-1c90-4f70-bb66-56f38d607ccb.png)

You have the following written in your Terminal app, where "your-name" is replaced with something specific to you:
`java --module-path /Users/your-name/Downloads/javafx-sdk-18.0.2/lib --add-modules ALL-MODULE-PATH -jar /Users/your-name/Downloads/mcaselector-2.1.jar`

If you are sure you have typed it in correctly, press return to start the app.

**Comment by Chrisspy36** [comment]
Thank you tons! Before I do this what happens if I get it wrong? Will I destroy my computer?

**Comment by Chrisspy36** [comment]
Okay I just followed the link you sent in the comment and there is not 18.0.2 It shows that it was created above but is not in the drop down menu. However 19 is available, do I need to update my normal java to 19 and then follow the same instructions? Thank you in advance, I'm really looking forward to hearing back from you!

**Comment by doinkythederp** [comment]
@Chrisspy36 Yeah, you'll need to update your Java to 19. Looks like they removed the Java 18 download.

**Comment by Chrisspy36** [comment]
Thank you for all your help. So one last problem I can't find java 19 that's available for download. Im on https://www.oracle.com/java/technologies/downloads/#java18 I can't seem to find a place here for 19, so my next question is if I just reinstall 17 will that be the program my computer uses? Or are there any extra steps I need to do to ensure its using 17 and not 18. I did have 17 before downloading 18 so if there is a way to select which version is operating? Then I can just run the fix as you first said using 17. Thank you so much for your time and help I really do appreciate it a lot!

**Comment by arvitus** [comment]
Java 19 is not out yet. Just use Java 17. Worked for me with javafx 19.
But if you want javafx 18.0.2 just tick the checkbox saying `Include older versions` and then you can select 18.0.2.
![grafik](https://user-images.githubusercontent.com/34866314/191123012-96545b12-2568-4de5-a65e-c1307ac1ba83.png)

**Comment by Chrisspy36** [comment]
@arvitus this is on the same website that @doinkythederp provided?

**Comment by doinkythederp** [comment]
Based on the screenshot, probably.

**Comment by arvitus** [comment]
@Chrisspy36 yes it is

**Comment by sosumee** [comment]
Thx for the help. worked for my M1 mini running macOS ventura with azul arm java 18

**Comment by Nukecraft5419** [comment]
Thank you very much I was looking for how to make it run on Fedora it worked I just had to adapt change the SDK part for Linux
